src/edu/washington/escience/myria/api/WorkerResource.java

```diff
@@ -0,0 +1,52 @@
+package edu.washington.escience.myria.api;
+
+import java.nio.file.Paths;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+import edu.washington.escience.myria.parallel.Worker;
+
+/**
+ * This is the class that handles API calls that return workers.
+ *
+ */
+@Path("/")
+@Produces(MyriaApiConstants.JSON_UTF_8)
+public final class WorkerResource {
+  /** The Myria server running on the worker. */
+  @Context private Worker worker;
+
+  /**
+   * @param qid query id.
+   * @return the stats of all hash tables of the given query.
+   */
+  @GET
+  @Path("/hashtable-query-{qid:\\d+}")
```

Comment:
I think a path format like `/query/<QID>/hashtable-stats` might be more readable.